### DISCUSSION TITLE
displaCy rendering visuals from right to left for languages such as Arabic

### DISCUSSION THREAD

**Discussion by SecroLoL** [explosion/spaCy#11061] [Help: Other Questions] [answered]
It seems that displaCy renders visuals for text in languages like Arabic from right to left, instead of left to right, leading to some unusual formatting. Is this intentional, or a bug? If not, is this because the language itself is meant to be read for right to left, like Arabic? This is the only non-bug explanation I can think of. I would prefer the text to be rendered from left to right, if possible.

Any help would be appreciated!

![Arabic Right to Left Rendering](https://user-images.githubusercontent.com/33791574/176780766-89ef8df0-3748-42ab-8fd1-31598f7f5b01.png)

The above image shows what I mean, where the image is being rendered on the right side of the screen and expands leftward.
The dependencies are not spaCy generated, so replicating the exact arcs needs to use the pipeline from Stanford's Stanza NLP library.

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @SecroLoL, thanks for reporting this. This is a bug related to `rtl` languages, and we're currently tracking it at https://github.com/explosion/spaCy/issues/4854. Your hunch is correct that it shows that way because Arabic is meant to be read from left to right. However, the rendering has somehow become dependent on that. In the meantime, if you're concerned with the presentation / the image appearing on the right side of the screen, perhaps it is possible to wrap the whole SVG / HTML in a `<div>` and manually moving it in CSS?
